internal/app/singularity/plugin_compile_linux.go

```diff
@@ -53,20 +69,38 @@ func CompilePlugin(sourceDir, destSif string) error {
 }
 
 // buildPlugin takes sourceDir which is the string path the host which contains the source code of
-// the plugin. The output path is where the plugin .so file should
-// end up.
+// the plugin. buildPlugin returns the path to the built file, along with an error
 //
 // This function essentially runs the `go build -buildmode=plugin [...]` command
-func buildPlugin(sourceDir string) error {
-	baseDir := filepath.Base(sourceDir)
-	scmd := exec.Command("singularity", "run", "--cleanenv", "-B",
-		sourceDir+":"+filepath.Join(containedSourceDir, baseDir),
-		containerURI, baseDir)
-
-	scmd.Stderr = os.Stderr
-	scmd.Stdout = os.Stdout
-	scmd.Stdin = os.Stdin
-	return scmd.Run()
+func buildPlugin(sourceDir string) (string, error) {
+	// assuming that sourceDir is within trimpath for now
+	out := pluginObjPath(sourceDir)
+	c := fmt.Sprintf(buildtmpl, trimpath, out, sourceDir)
+	buildcmd := exec.Command("go", strings.Split(c, " ")...)
```

Comment:
I haven't had a chance to go thru this, but I have the hunch that this is missing build tags. I think when building singularity we are passing a series of build tags, and I assume it's possible to tweak those. I'm guessing the plugin needs to be built with the same set of tags.

Reply:
Actually, the plugins don't need to be built with the same tags. You can verify this by compiling and installing Singularity, then compiling the example plugin using `singularity plugin compile $(GOPATH)/src/github.com/sylabs/singularity/examples/plugins/test-plugin` and then doing a `sudo cp test-plugin.so $(PREFIX)/libexecdir/plugin`. You'll see that in the current state, singularity is compiled with up to a few tags, and even though the plugin isn't it will still work

Reply:
That's surprising. I'm guessing the build flags we are using are not changing the exposed API but instead they only changing the implementation of some functions. Either we make that a promise (build tags won't change the API) or we make sure plugins are built with the same tags.

We can punt this for now.